app/controllers/kudos_controller.rb

```diff
@@ -15,8 +21,10 @@ def create
     @kudo = Kudo.new(kudo_params)
     @kudo.giver_id = current_employee.id
     if @kudo.save
-      @current_employee.number_of_available_kudos -= 1
-      @current_employee.save
+      Employee.transaction do
+        @current_employee.number_of_available_kudos -= 1
+        @current_employee.save
```

Comment:
Hej ta transkacje nic nie wnosi do kodu.
transakcje wykonujemy aby mieć pewność że wszystkie zapisy do bazy danych się udały poprawnie, nie chcemy mieć sytuacji że ktoś da kudo, ale nie odejmiemy mu `number_of_available_kudos`. Ponadto trasakcja jest rollbackowana tylko jeżeli wystąpi w niej nieobsłużony error -> zapoznaj się z [bang methods w activerecord](https://riptutorial.com/ruby-on-rails/example/9285/activerecord-bang-----methods)

Reply:
I should move it right berfore the if statement shouldn't I ? :)